protocol/x/clob/keeper/process_operations.go

```diff
@@ -619,27 +619,43 @@ func (k Keeper) PersistMatchDeleveragingToState(
 	matchDeleveraging *types.MatchPerpetualDeleveraging,
 ) error {
 	liquidatedSubaccountId := matchDeleveraging.GetLiquidated()
+	perpetualId := matchDeleveraging.GetPerpetualId()
+	clobPair := k.mustGetClobPairForPerpetualId(ctx, perpetualId)
 
-	// Validate that the provided subaccount can be deleveraged.
-	if canDeleverageSubaccount, err := k.CanDeleverageSubaccount(ctx, liquidatedSubaccountId); err != nil {
-		panic(
-			fmt.Sprintf(
-				"PersistMatchDeleveragingToState: Failed to determine if subaccount can be deleveraged. "+
-					"SubaccountId %+v, error %+v",
-				liquidatedSubaccountId,
-				err,
-			),
-		)
-	} else if !canDeleverageSubaccount {
-		// TODO(CLOB-853): Add more verbose error logging about why deleveraging failed validation.
+	// If `IsFinalSettlement` flag on deleveraging match is set to true, verify that the market is in final settlement.
+	if matchDeleveraging.IsFinalSettlement && clobPair.Status != types.ClobPair_STATUS_FINAL_SETTLEMENT {
 		return errorsmod.Wrapf(
-			types.ErrInvalidDeleveragedSubaccount,
-			"Subaccount %+v failed deleveraging validation",
-			liquidatedSubaccountId,
+			types.ErrFinalSettlementDeleveragingMatchForActiveMarket,
+			"MatchPerpetualDeleveraging %+v is a final settlement match, but the clob pair %+v is not in final settlement",
+			matchDeleveraging,
+			clobPair,
 		)
 	}
 
-	perpetualId := matchDeleveraging.GetPerpetualId()
+	// Skip the below validation for final settlement deleveraging matches. Final settlement matches do
+	// not require the subaccount to have negative TNC to be considered deleveragable.
+	if !matchDeleveraging.IsFinalSettlement {
```

Comment:
only need to verify that the subaccount has negative TNC if this is not a final settlement match